buf/registry/plugin/v1beta1/plugin.proto

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.registry.plugin.v1beta1;
+
+import "buf/registry/plugin/v1beta1/collection.proto";
+import "buf/registry/priv/extension/v1beta1/extension.proto";
+import "buf/validate/validate.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/plugin/v1beta1";
+
+message Plugin {
+  option (buf.registry.priv.extension.v1beta1.message).response_only = true;
+
+  // The id of the Plugin.
+  string id = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The time the Plugin was created on the BSR.
+  google.protobuf.Timestamp create_time = 2 [(buf.validate.field).required = true];
+  // The last time the Plugin was updated on the BSR.
+  google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
+  // The name of the Plugin.
+  //
+  // Unique within a given Organization.
+  string name = 4 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 100
+  }];
+  // The id of the Organization that owns the Plugin.
+  string owner_id = 5 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The Plugin's visibility, either public or private.
+  PluginVisibility visibility = 6 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin state, either active or deprecated.
+  PluginState state = 7 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin type.
+  PluginType type = 8 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The configurable short description of the Plugin.
+  string description = 9 [(buf.validate.field).string.max_len = 350];
```

Comment:
Why would I want this in addition to the commit-level documentation?

Reply:
Perhaps we could remove both. Do you have a sense of how many descriptions are set for modules? If not many, I rather just wipe them all.

Reply:
It's a _very_ small amount. It's also not directly surfaced in the BSR outside of the repository settings page.

Isn't this a useful field that allows users to describe their module or plugin as a whole?

Reply:
In a `doc.short` and `doc.long` world (which I think Modules could migrate to as well), `doc.short becomes `description`.

To enable this easily, I'd be in favor of re-defining `Doc` on `bufplugin` to make `short` and `long` distinct, ie you do *not* do `doc.short + "\n\n" + doc.long`, instead they are completely disjoint. In this world, for modules, `README.md/buf.md` maps to `doc.long`, `description` maps to `doc.short`.

BSR outside of repo settings surfaces `doc.short` of latest commit on default label.

Cc @perezd

Reply:
There's a bit of a discrepancy in how these work?

- The module readme (long) is on the commit. The description (short) is on the module entity.
- The plugin long and short are both on the commit.

Would users update the module description via the UI/API, and the plugin description by pushing a commit?

To make things work equally, we could ignore plugin info `doc.short` and only store `doc.long`. Leaving long descriptions on the commit, and short descriptions on the module/plugin entity.

Reply:
I don't understand the proposal you just outlined - to be clear on mine:

- Delete `Module.description` entirely.
- Add `Doc` to both the module and plugin packages.
- Make it so that `doc.short` is not concat'ed with `doc.long` in the Bufplugin API, ie if you want the content of `doc.short` to be shown in a long description of a plugin, it must be copied to `doc.long` explicitly.
- Delete `Plugin.description` in this PR.
- Derive the Module/Plugin description in the BSR UI from the `doc.short` field of the commit on the default label.
- Derive the `README.md` from `doc.long`.

Reply:
Alternatively, we could kill `Doc` entirely, and instead have rields `description` and `documentation` added to https://github.com/bufbuild/bufplugin/blob/main/buf/plugin/info/v1/plugin_info.proto#L37 and `module.v1.Commit/plugin.v1.Commit`.

Reply:
Talked offline:

- `source_url` and `description` to live on `Module` and `Plugin`.
- README and LICENSE will live on the `Commit` level for both `Module` and `Plugin`.

See https://github.com/bufbuild/bufplugin/pull/7